packages/fury-adapter-swagger/lib/parser.js

```diff
@@ -325,6 +325,22 @@ class Parser {
           });
         }
 
+        if (this.swagger.info.termsOfService) {
+          this.withPath('termsOfService', () => {
+            const { Link } = this.namespace.elements;
+            const link = new Link();
+
+            link.relation = 'terms-of-service';
+            link.href = this.swagger.info.termsOfService;
+
+            if (this.generateSourceMap) {
+              this.createSourceMap(this.api.meta.get('termsOfService'), this.path);
```

Comment:
```suggestion
              this.createSourceMap(link, this.path);
```

We haven’t pushed the link into `this.api` yet so the source map wouldn’t end up in the right place. I think some of the other branches above such as for `title` might be broken. I’ll confirm when I’m back on Monday and advise how we can write test case for the source maps.

Reply:
going to change, mind that if I comment out the whole 3 lines

```js
// if (this.generateSourceMap) {
//   this.createSourceMap(link, this.path);
// }
```

the test is still passing

Reply:
I've created https://github.com/apiaryio/api-elements.js/pull/412 which we can rebase this PR onto once merged. Then you can have a test fixture named `info-termsOfService.sourcemap.json` which is similiar to `info-termsOfService.json` except also contains source maps.